pangea-sdk/v3/service/vault/symmetric.go

```diff
@@ -97,3 +97,96 @@ type EncryptStructuredResult struct {
 	// Structured data with filtered fields encrypted/decrypted.
 	StructuredData map[string]interface{} `json:"structured_data"`
 }
+
+type TransformAlphabet string
+
+const (
+	TAalphalower        TransformAlphabet = "alphalower"        // Lowercase alphabet (a-z).
+	TAalphanumeric      TransformAlphabet = "alphanumeric"      // Alphanumeric (a-z, A-Z, 0-9).
+	TAalphanumericlower TransformAlphabet = "alphanumericlower" // Lowercase alphabet with numbers (a-z, 0-9).
+	TAalphanumericupper TransformAlphabet = "alphanumericupper" // Uppercase alphabet with numbers (A-Z, 0-9).
+	TAalphaupper        TransformAlphabet = "alphaupper"        // Uppercase alphabet (A-Z).
+	TAnumeric           TransformAlphabet = "numeric"           // Numeric (0-9).
+)
+
+// Parameters for an encrypt transform request.
+type EncryptTransformRequest struct {
+	pangea.BaseRequest
+
+	// The ID of the key to use.
+	ID string `json:"id"`
+
+	// Message to be encrypted.
+	PlainText string `json:"plain_text"`
+
+	// Set of characters to use for format-preserving encryption (FPE).
+	Alphabet TransformAlphabet `json:"alphabet"`
+
+	// User provided tweak string. If not provided, a random string will be
+	// generated and returned. The user must securely store the tweak source
+	// which will be needed to decrypt the data.
+	Tweak *string `json:"tweak,omitempty"`
+
+	// The item version. Defaults to the current version.
+	Version *int `json:"version,omitempty"`
+}
+
+// Result of an encrypt transform request.
+type EncryptTransformResult struct {
+	// The item ID.
+	ID string `json:"id"`
+
+	// The encrypted message.
+	CipherText string `json:"cipher_text"`
+
+	// The item version.
+	Version int `json:"version"`
+
+	// The algorithm of the key.
+	Algorithm string `json:"algorithm"`
+
+	// User provided tweak string. If not provided, a random string will be
+	// generated and returned. The user must securely store the tweak source
+	// which will be needed to decrypt the data.
+	Tweak *string `json:"tweak,omitempty"`
```

Comment:
Do we need a pointer here? It should be always present

Reply:
Good catch, done.